tests/backup/backup_test.go

```diff
@@ -1011,467 +1025,765 @@ var _ = Describe("{BackupCrashVolDriver}", func() {
+						namespace, restoreName), func() {
 
-func DeleteBucket(provider string, bucketName string) {
-	Step(fmt.Sprintf("Delete bucket [%s]", bucketName), func() {
-		switch provider {
-		// TODO(stgleb): PTX-2359 Add DeleteAzureBucket
-		case drivers.ProviderAws:
-			DeleteS3Bucket(bucketName)
-		case drivers.ProviderAzure:
-			DeleteAzureBucket(bucketName)
+						ctx, err := backup.GetPxCentralAdminCtx()
+						Expect(err).NotTo(HaveOccurred(),
+							fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]",
+								err))
+						err = Inst().Backup.WaitForRestoreCompletion(ctx, restoreName, orgID,
+							backupRestoreCompletionTimeoutMin*time.Minute,
+							retrySeconds*time.Second)
+						if err != nil {
+							bkpNamespaceErrors[namespace] = err
+							logrus.Errorf("Failed to wait for restore [%s] to complete. Error: [%v]",
+								restoreName, err)
+						}
+					})
+				}(&wg, namespace, restoreName)
+			}
 		}
-	})
-}
-
-func DeleteS3Bucket(bucketName string) {
-	id, secret, endpoint, s3Region, disableSSLBool := getAWSDetailsFromEnv()
-	sess, err := session.NewSession(&aws.Config{
-		Endpoint:         aws.String(endpoint),
-		Credentials:      credentials.NewStaticCredentials(id, secret, ""),
-		Region:           aws.String(s3Region),
-		DisableSSL:       aws.Bool(disableSSLBool),
-		S3ForcePathStyle: aws.Bool(true),
-	},
-	)
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to get S3 session to create bucket. Error: [%v]", err))
+		wg.Wait()
 
-	S3Client := s3.New(sess)
+		// Change namespaces to restored apps only after backed up apps are cleaned up
+		// to avoid switching back namespaces to backup namespaces
+		Step("Validate Restored applications", func() {
+			destClusterConfigPath, err := getDestinationClusterConfigPath()
+			Expect(err).NotTo(HaveOccurred(),
+				fmt.Sprintf("Failed to get kubeconfig path for destination cluster. Error: [%v]", err))
 
-	iter := s3manager.NewDeleteListIterator(S3Client, &s3.ListObjectsInput{
-		Bucket: aws.String(bucketName),
-	})
+			SetClusterContext(destClusterConfigPath)
 
-	err = s3manager.NewBatchDeleteWithClient(S3Client).Delete(aws.BackgroundContext(), iter)
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Unable to delete objects from bucket %q, %v", bucketName, err))
+			// Populate contexts
+			for _, ctx := range contexts {
+				ctx.SkipClusterScopedObject = true
+				ctx.SkipVolumeValidation = true
+			}
 
-	_, err = S3Client.DeleteBucket(&s3.DeleteBucketInput{
-		Bucket: aws.String(bucketName),
-	})
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to delete bucket [%v]. Error: [%v]", bucketName, err))
-}
+			ValidateRestoredApplicationsGetErr(contexts, volumeParams, bkpNamespaceErrors)
+		})
 
-func DeleteAzureBucket(bucketName string) {
-	// From the Azure portal, get your Storage account blob service URL endpoint.
-	_, _, _, _, accountName, accountKey := getAzureCredsFromEnv()
+		successfulRestores = len(bkpNamespaces) - len(bkpNamespaceErrors)
```

Comment:
Same as backup, we might want to collect the failed restores and their associated errors.